### PR TITLE
PRMDR-565 - Download manifest polling and gateway methods refactor

### PR DESCRIPTION
**Pull request by abbas-khan10**

### CODE REVIEW

File: infrastructure/dynamo_db.tf [hunk 1]
```diff
@@ -96,25 +96,34 @@ module "zip_store_reference_dynamodb_table" {
...
     {
       name            = "NhsNumberIndex"
       hash_key        = "NhsNumber"
       projection_type = "ALL"
+    },
+    {
+      name            = "FileLocationsIndex"
+      hash_key        = "FileLocation"
+      projection_type = "ALL"
     }
```
Comment (joefong-nhs) [inline infrastructure/dynamo_db.tf:127]: I think we could remove the FileLocationsIndex GSI, as we probably don't have a use case for doing a query with the FileLocation string

File: infrastructure/dynamo_db.tf [hunk 2]
```diff
@@ -96,26 +96,26 @@ module "zip_store_reference_dynamodb_table" {
...
     {
-      name = "FileLocation"
+      name = "JobId"
       type = "S"
     },
     {
       name = "NhsNumber"
       type = "S"
-    }
+    },
```
Comment (NogaNHS) [inline infrastructure/dynamo_db.tf:105]: We don't use nhs number in that table anymore

File: infrastructure/dynamo_db.tf [hunk 3]
```diff
@@ -96,26 +96,26 @@ module "zip_store_reference_dynamodb_table" {
...
-      hash_key        = "FileLocation"
+      name            = "JobIdIndex"
+      hash_key        = "JobId"
       projection_type = "ALL"
     },
     {
       name            = "NhsNumberIndex"
       hash_key        = "NhsNumber"
       projection_type = "ALL"
-    }
```
Comment (NogaNHS) [inline infrastructure/dynamo_db.tf:118]: Also not needed

File: infrastructure/lambda-bulk-upload.tf [hunk 4]
```diff
@@ -33,8 +33,6 @@ module "bulk-upload-lambda" {
 
   is_gateway_integration_needed  = false
   is_invoked_from_gateway        = false
-  memory_size                    = 512
-  lambda_timeout                 = 900
```
Comment (NogaNHS) [inline infrastructure/lambda-bulk-upload.tf:37]: Why are we reducing the timeout/

File: infrastructure/lambda-generate-document-manifest.tf [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  source                   = "./modules/lambda"
+  name                     = "GenerateDocumentManifest"
+  handler                  = "handlers.generate_document_manifest_handler.lambda_handler"
+  lambda_timeout           = 900
+  lambda_ephemeral_storage = 512
+  memory_size              = 512
+  iam_role_policies = [
+    module.document_reference_dynamodb_table.dynamodb_policy,
+    module.ndr-document-store.s3_object_access_policy,
+    module.lloyd_george_reference_dynamodb_table.dynamodb_policy,
```
Comment (NogaNHS) [inline infrastructure/lambda-generate-document-manifest.tf:53]: we don't need `module.lloyd_george_reference_dynamodb_table.dynamodb_policy` or `module.document_reference_dynamodb_table.dynamodb_policy,`

File: infrastructure/lambda-generate-document-manifest.tf [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    module.ndr-lloyd-george-store.s3_object_access_policy,
+    module.zip_store_reference_dynamodb_table.dynamodb_policy,
+    module.ndr-zip-request-store.s3_object_access_policy,
+    "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
+    "arn:aws:iam::aws:policy/CloudWatchLambdaInsightsExecutionRolePolicy",
+    module.ndr-app-config.app_config_policy_arn,
```
Comment (NogaNHS) [inline infrastructure/lambda-generate-document-manifest.tf:59]: Do we need config?

File: infrastructure/lambda-generate-document-manifest.tf [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+  rest_api_id       = null
+  api_execution_arn = null
+  lambda_environment_variables = {
+    APPCONFIG_APPLICATION        = module.ndr-app-config.app_config_application_id
+    APPCONFIG_ENVIRONMENT        = module.ndr-app-config.app_config_environment_id
+    APPCONFIG_CONFIGURATION      = module.ndr-app-config.app_config_configuration_profile_id
+    DOCUMENT_STORE_BUCKET_NAME   = "${terraform.workspace}-${var.docstore_bucket_name}"
+    DOCUMENT_STORE_DYNAMODB_NAME = "${terraform.workspace}_${var.docstore_dynamodb_table_name}"
+    LLOYD_GEORGE_BUCKET_NAME     = "${terraform.workspace}-${var.lloyd_george_bucket_name}"
+    LLOYD_GEORGE_DYNAMODB_NAME   = "${terraform.workspace}_${var.lloyd_george_dynamodb_table_name}"
```
Comment (NogaNHS) [inline infrastructure/lambda-generate-document-manifest.tf:71]: We don't use those

File: infrastructure/lambda-generate-document-manifest.tf [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    ZIPPED_STORE_DYNAMODB_NAME   = "${terraform.workspace}_${var.zip_store_dynamodb_table_name}"
+    SPLUNK_SQS_QUEUE_URL         = try(module.sqs-splunk-queue[0].sqs_url, null)
+    WORKSPACE                    = terraform.workspace
+    PRESIGNED_ASSUME_ROLE        = aws_iam_role.manifest_presign_url_role.arn
+  }
+  is_gateway_integration_needed = false
+  is_invoked_from_gateway       = false
+
+  depends_on = [
+    aws_iam_policy.lambda_audit_splunk_sqs_queue_send_policy[0],
```
Comment (NogaNHS) [inline infrastructure/lambda-generate-document-manifest.tf:82]: Add depends on `    module.zip_store_reference_dynamodb_table,
    module.ndr-zip-request-store`

**Review by NogaNHS — APPROVED** [review]

**Pull request merged**